File: PrivateVault.ai-main/sdk/python/privatevault_sdk/quorum.py
from privatevault_sdk.client import Client


def validate_quorum(client: Client, action: str, payload: dict, approvals: list, tenant_id: str | None = None):
    body = {"action": action, "payload": payload, "approvals": approvals}
    if tenant_id:
        body["tenant_id"] = tenant_id
    resp = client.request("POST", "/quorum/validate", json_body=body)
    resp.raise_for_status()
    return resp.json()

File: PrivateVault.ai-main/sdk/node/src/quorum.js
export async function validateQuorum(client, { action, payload, approvals, tenantId }) {
  const body = { action, payload, approvals };
  if (tenantId) {
    body.tenant_id = tenantId;
  }
  const resp = await client.request('POST', '/quorum/validate', { jsonBody: body });
  if (!resp.ok) {
    throw new Error(`quorum validate failed: ${resp.status}`);
  }
  return resp.json();
}
